File: app/src/main/java/com/example/seki/android_custom_linklifytext_sample/ClickableTextPresenter.java
package com.example.seki.android_custom_linklifytext_sample;

import android.text.Spannable;

import com.example.seki.android_custom_linklifytext_sample.clickableString.ClickableString;

public class ClickableTextPresenter {

    final View mView;

    public interface View {
        void setPresenter(ClickableTextPresenter presenter);
        void addData(Spannable text);
        void clearInputText();
        void showTextEmptyMessage();
    }

    public ClickableTextPresenter(View v) {
        mView = v;
        mView.setPresenter(this);
    }

    public void onClickSubmit(String clickableStrings, String text) {
        if (text.isEmpty()) {
            mView.showTextEmptyMessage();
            mView.clearInputText();
            return;
        }

        mView.addData(ClickableString.create(clickableStrings, text));
        mView.clearInputText();
    }
}

File: app/src/main/java/com/example/seki/androidCustomLinklifytextSample/ClickableTextPresenter.kt
package com.example.seki.androidCustomLinklifytextSample;

import android.text.Spannable;

import com.example.seki.androidCustomLinklifytextSample.clickableString.ClickableString

class ClickableTextPresenter(v: View) {

    private var mView: View? = null

    interface View {
        fun setPresenter(presenter: ClickableTextPresenter)
        fun addData(text: Spannable)
        fun clearInputText()
        fun showTextEmptyMessage()
    }

    init {
        mView = v
        mView!!.setPresenter(this)
    }

    fun onClickSubmit(clickableStrings: String, text: String) {
        if (text.length <= 0) {
            mView!!.showTextEmptyMessage()
            mView!!.clearInputText()
            return
        }

        mView!!.addData(ClickableString.create(clickableStrings, text))
        mView!!.clearInputText()
    }
}
